src/monodroid/jni/cpp-util.hh

```diff
@@ -112,7 +112,7 @@ namespace xamarin::android
 		// `sizeof... (Length)` part which subtracts the number of template parameters - the amount of NUL bytes so that
 		// we don't waste space.
 		constexpr size_t total_length = (... + Length) - sizeof... (Length);
-		char_array<total_length + 1> ret;
+		char_array<total_length + 1> ret; // lgtm [cpp/paddingbyteinformationdisclosure] the buffer is filled in the loop below
```

Comment:
The docs warn about "uninitialized padding bytes", which I *imagine* could happen if we had:

```cpp
auto a = concat_const ("foo", "bar", "baz");
auto b = concat_const ("qux", "quux");
```

wherein sizeof(a) is 10, which *won't* be stack aligned, necessitating padding between `a` and `b`.

But I also wonder if I'm overthinking that example?

Additionally, the only places I see `concat_const()` being used are for `static constexpr auto` members, which are *not* local variables, so any padding between static members would by definition be zero-initialized…

But if that *weren't* the case, and we *were* using this for local variables…  Could this be a problem?  I'm still dubious, but I'm not completely certain.

Reply:
I think this is about the `+ 1` byte. `total_length` is a compile-time constant computed without counting the trailing `\0` bytes of input strings, and so the array length "should" be `total_length` from the POV of the analyzer - the extra byte is seen as uninitialized "padding", while it is a `\0` C char termination byte in reality, set one line below this one. 

I think the analyzer gets confused by the template packs usage in both the declaration and the loop below (or maybe it simply doesn't analyze the loop at all). Whichever is the case, it's not a concern and not a bug, so we can hush it without worries.